tests/otel_integrations/test_fastapi.py

```diff
@@ -1508,3 +1512,360 @@ def test_scrubbing(client: TestClient, exporter: TestExporter) -> None:
             },
         ]
     )
+
+
+def make_request_hook_spans(record_send_receive: bool):
+    # Instrument an app with request hooks to make sure that they work.
+    # We make a new app here instead of using the fixtures because the same app can't be instrumented twice.
+    # Then make a request to it to generate spans.
+    # The tests then check that the spans are in the right place both with and without send/receive spans.
```

Comment:
This is what the spans look like:

<img width="289" alt="Screenshot 2024-08-08 at 13 45 09" src="https://github.com/user-attachments/assets/90a68901-9dbb-4a28-8886-3a2e4139f2db">